confirmation_height_clear cli account param

JerzyStanislawski opened the pull request:
Make account argument required for confirmation_height_clear cli command.
Issue reference: https://github.com/nanocurrency/nano-node/issues/3831
resolves #3831

JerzyStanislawski committed:
confirmation_height_clear cli account param

Commit:
Add back missing password option

thsfs commented:
Sent a fix to the line I mentioned earlier, there is also the `password` option for `account_create`, not related to `confirmation_height_clear`, but changed in the first commit.

thsfs commented:
@dsiganos , could you double check and get this merged if you approve?

dsiganos commented on nano/node/cli.cpp:
```diff
@@ -609,17 +608,22 @@ std::error_code nano::handle_node_options (boost::program_options::variables_map
-				auto transaction (node.node->store.tx_begin_write ());
-				reset_confirmation_heights (transaction, node.node->network_params.ledger, node.node->store);
-				std::cout << "Confirmation heights of all accounts (except genesis which is set to 1) are set to 0" << std::endl;
+				std::cerr << "confirmation_height_clear command requires one <account> option\n";
```
It could also say: "or 'any' to clear all accounts.

thsfs replied:
Improved this and also the help message to the `--confirmation_height_clear` option.

Commit:
Improve the error/help messages to the required account option

- Improve the error message to say the account can be 'all'
- Improve the confirmation_height_clear help message to inform the value
'all' can be passed to clear all accounts

thsfs commented:
Thanks @JerzyStanislawski !
